### PR TITLE
Reduce ax-8 usage

### PR DESCRIPTION
**Pull request by GinoGiotto**
A few more axiom usage revisions, specifically:

* Edit [rzal](https://us.metamath.org/mpeuni/rzal.html) -> avoid df-clel, ax-8
* Edit [rexn0](https://us.metamath.org/mpeuni/rexn0.html) -> avoid df-clel, ax-8
* Edit [ralidm](https://us.metamath.org/mpeuni/ralidm.html) -> avoid df-clel, df-cleq, ax-8, ax-9, ax-ext
* Edit [ral0](https://us.metamath.org/mpeuni/ral0.html) -> avoid df-clel, ax-8
* Edit [ralf0](https://us.metamath.org/mpeuni/ralf0.html) -> avoid df-clel, ax-8

Note: I used `(/) = { x | F. }` in the proofs of [rzal](https://us.metamath.org/mpeuni/rzal.html) and [ralf0](https://us.metamath.org/mpeuni/ralf0.html).

Axiom usage here: https://github.com/metamath/set.mm/commit/4f41272666476da49a4ae56275ce1fa8b49d4e9c

### CODE REVIEW

**Review by tirix — APPROVED** [review]

**Comment by benjub** [comment]
> Note: I used `(/) = { x | F. }` in the proofs of [rzal](https://us.metamath.org/mpeuni/rzal.html) and [ralf0](https://us.metamath.org/mpeuni/ralf0.html).

Feel free to move https://us.metamath.org/mpeuni/bj-df-nul.html to Main, as, e.g., dfnul4.

**Comment by GinoGiotto** [comment]
> Feel free to move [us.metamath.org/mpeuni/bj-df-nul.html](https://us.metamath.org/mpeuni/bj-df-nul.html) to Main, as, e.g., dfnul4.

Done. I changed its proof to avoid ax-8.

File: set.mm [hunk 1]
```diff
@@ -39637,6 +39637,12 @@ among classes ( ~ eq0 , as opposed to the weaker uniqueness among sets,
     ( weq wn cab cv wcel wa crab pm3.24 equid 2th con1bii dfnul2 df-rab 3eqtr4i
     c0 abbii ) AACZDZAEAFBGZUADZHZAEQUBABITUCAUCSUCDSUAJAKLMRANUBABOP $.
 
+  $( Alternate definition of the empty class/set.  (Contributed by BJ,
+     30-Nov-2019.) $)
```
Comment (benjub) [inline set.mm:39641]: Probably credit is due to you, here. Something like `Avoid ~ ax-8 , ~ df-clel .  (Revised by GG, ...`

File: set.mm [hunk 2]
```diff
@@ -41110,37 +41115,96 @@ disjoint classes (see ~ disjdif ).  Part of proof of Corollary 6K of
...
+      USUTVAVEVIVODVBVCABCVFVD $.
+    $( $j usage 'ralf0' avoids 'df-clel' 'ax-8'; $)
+  $}
+
+  ${
+    $d x A $.
+    $( Obsolete version of ~ rzal as of 2-Sep-2024.  (Contributed by NM,
+       11-Mar-1997.)  (Proof shortened by Andrew Salmon, 26-Jun-2011.)
+       (Proof modification is discouraged.)  (New usage is discouraged.) $)
+    rzalOLD $p |- ( A = (/) -> A. x e. A ph ) $=
```
Comment (benjub) [inline set.mm:41176]: I would say "ALT" since much shorter ?

**Review by benjub — CHANGES_REQUESTED** [review]

**Review by icecream17 — APPROVED** [review]
ab0 can be shortened by expanding df-nul (or dfnul4) and using abbi (no cost to axioms)

I can't tell right now, but there's a chance this could be useful to shorten one of the theorems in this pr

**Comment by benjub** [comment]
> ab0 can be shortened by expanding df-nul (or dfnul4) and using abbi (no cost to axioms)
> 
> I can't tell right now, but there's a chance this could be useful to shorten one of the theorems in this pr

Can you do that in a separate PR ?

**Comment by GinoGiotto** [comment]
> ab0 can be shortened by expanding df-nul (or dfnul4) and using abbi (no cost to axioms)

In general, I have a preference for ~abbii over ~abbi, because ~abbi requires ax-10, ax-11, ax-12 and it seems none of them can be removed (I've attempted it in the past without success). This makes it a theorem with rather limited applications, but it's fine in particular cases like ~ab0 since it uses those axioms already.

**Review by benjub — APPROVED** [review]

**Pull request merged**